ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/GlobalVariable.java

```diff
@@ -31,20 +31,37 @@
  */
 public class GlobalVariable {
     private final String type;
+    private final String kind;
     private final QuotedIdentifier variableName;
     private final boolean isAssignableToAny;
     private final String qualifiersAndMetadata;
+    private final boolean isAssignedWithVar;
 
-    public GlobalVariable(String type, QuotedIdentifier variableName,
+    private static final String VAR_TYPE_DESC = "VAR_TYPE_DESC";
+
+    public GlobalVariable(String type, String typeLHS, String kind, QuotedIdentifier variableName,
                           boolean isAssignableToAny, String qualifiersAndMetadata) {
         this.type = Objects.requireNonNull(type);
         this.isAssignableToAny = isAssignableToAny;
         this.qualifiersAndMetadata = Objects.requireNonNull(qualifiersAndMetadata);
         this.variableName = variableName;
+        isAssignedWithVar = typeLHS.equals(VAR_TYPE_DESC) ? true : false;
+        this.kind = kind;
     }
 
     public String getType() {
-        return type;
+        return this.type;
+    }
+
+    public String getKind() {
+        return this.kind;
+    }
+
+    public String getTypeLHS() {
+        if (isAssignedWithVar) {
+            return VAR_TYPE_DESC;
+        }
+        return this.type;
     }
```

Comment:
Isn't this separate method unnecessary? Why can't `getType()` itself be used for this?

Reply:
Removed in 3fc03c0dee2d7091929456e76b603ab332cb786e